### PR TITLE
CI link-checking + fixed links

### LINKED ISSUE #1435: [FEAT] Test out link checking in docs
### Is your proposal related to a problem?

Tracking when links break in the docs is annoying - automating it would make life much easier.

### Describe the solution you'd like

Test out https://pypi.org/project/mkdocs-linkcheck/ and (if successful) add to the lining GH action.

### PR DESCRIPTION
**Pull request by ADBond**
### Type of PR

- [ ] BUG
- [ ] FEAT
- [x] MAINT
- [x] DOC

### Is your Pull Request linked to an existing Issue or Pull Request?

Closes #1435

### Give a brief description for the solution you have provided

This adds [lychee](https://github.com/lycheeverse/lychee) link-checker to the docs action. As such it also breaks up the steps into separate build + push.

I have also made changes to get this action passing, namely:
* fixing broken links directly
* removing some `ipynb` output for embedded dashboards, which are anyway currently broken on the live site
* added some logic to preprocessing script to fix the links in `CONTRIBUTING.md`

The latter point is not ideal, but doesn't seem _massively_ worse than the fudges we already have for markdown includes. Could probably similarly be addressed by #1395

Action is tested on my fork, see:
* [a failing run](https://github.com/ADBond/splink/actions/runs/7697613490)
* [successful run](https://github.com/ADBond/splink/actions/runs/7697926883)
* [the current successful build](https://github.com/ADBond/splink/tree/gh-pages)

### PR Checklist

- [ ] Added documentation for changes
- [ ] Added feature to example notebooks or tutorial (if appropriate)
- [ ] Added tests (if appropriate)
- [ ] Updated CHANGELOG.md (if appropriate)
- [x] Made changes based off the latest version of Splink
- [ ] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

**Review by RossKen — COMMENTED** [review]
Hey @ADBond - thanks for this PR. Lychee looks like a great solution for this!

All looks to work as expected from my end 🎉

One workflow thing that came to mind. In this version the links only get checked in our mkdocs action which is only triggered after merging, right? Therefore any issues flagged will have to be fixed in a new PR with another review etc. Is this the best solution? I guess the alternative is to run a docs build and link check at the PR stage as well as deploy, but I am wary of adding more time to the actions on PRs after all of the effort to make tests run quicker.

**Comment by ADBond** [comment]
> One workflow thing that came to mind. In this version the links only get checked in our mkdocs action which is only triggered after merging, right? Therefore any issues flagged will have to be fixed in a new PR with another review etc. 

Ah yeah, good point! I think it would be good to have something to give feedback before merging. There are a few options we could use as a trigger (so that it won't run on _every_ PR):
* run only on PRs that change files in `docs/` (and other relevant files?)
* run only on PRs with a label of 'documentation'
* run on PRs with specific words in title (e.g. starts with [DOC]) or has specific text in the body
* allow it to be manually triggered (either from actions menu, or e.g. using a specific phrase in a PR comment 'check my links')

I guess probably the first option would make most sense as it doesn't require anything active from the submitter - it should 'just work', but curious if you have any thoughts.

This will take a little bit of fiddling, as would need to split the 'deploy' job into separate 'build' and 'deploy' jobs so that when we run in a PR we don't actually publish the results until it is merged. So I will probably do this in a separate PR - depending on your thoughts can either merge this as-is and follow up with the improved workflow, or open a PR into this branch.

**Review by RossKen — APPROVED** [review]
Super stuff - thanks @ADBond! 👏

**Pull request merged**
